diff -u -r a/mon-put-instance-data.pl b/mon-put-instance-data.pl
--- a/mon-put-instance-data.pl	2015-03-06 21:57:08.000000000 +0100
+++ b/mon-put-instance-data.pl	2016-05-04 15:20:16.000000000 +0200
@@ -373,8 +373,8 @@
 {
   my %opts = ();
   $opts{'aws-credential-file'} = $aws_credential_file;
-  $opts{'aws-access-key-id'} = $aws_access_key_id;
-  $opts{'aws-secret-key'} = $aws_secret_key;
+  $opts{'aws-access-key-id'} = $aws_access_key_id || $ENV{'AWS_ACCESS_KEY_ID'};
+  $opts{'aws-secret-key'} = $aws_secret_key || $ENV{'AWS_SECRET_ACCESS_KEY'};
   $opts{'verbose'} = $verbose;
   $opts{'verify'} = $verify;
   $opts{'user-agent'} = "$client_name/$version";
@@ -579,8 +579,8 @@
 {
   my %opts = ();
   $opts{'aws-credential-file'} = $aws_credential_file;
-  $opts{'aws-access-key-id'} = $aws_access_key_id;
-  $opts{'aws-secret-key'} = $aws_secret_key;
+  $opts{'aws-access-key-id'} = $aws_access_key_id || $ENV{'AWS_ACCESS_KEY_ID'};
+  $opts{'aws-secret-key'} = $aws_secret_key || $ENV{'AWS_SECRET_ACCESS_KEY'};
   $opts{'retries'} = 2;
   $opts{'verbose'} = $verbose;
   $opts{'verify'} = $verify;
